src/main/java/com/teragrep/cfe_39/consumers/kafka/HDFSWrite.java

```diff
@@ -0,0 +1,250 @@
+            hdfsuri = config.getHdfsuri(); // Get from config.
+
+            // The filepath should be something like hdfs:///opt/teragrep/cfe_39/srv/topic_name/0.12345 where 12345 is offset and 0 the partition.
+            // In other words the directory named topic_name holds files that are named and arranged based on partition and the partition's offset. Every partition has its own set of unique offset values.
+            // The values are fetched from config and input parameters (topic+partition+offset).
+            path = config.getHdfsPath() + "/" + lastObject.topic; // directory path is constructed from HdfsPath and topic name.
+            fileName = lastObject.partition + "." + lastObject.offset; // filename should be constructed from partition and offset.
+
+
+            // Set HADOOP user here, Kerberus parameters most likely needs to be added here too.
+            // System.setProperty("HADOOP_USER_NAME", "hdfs"); // Not needed because user authentication is done by kerberos?
+            // System.setProperty("hadoop.home.dir", "/"); // Not needed because user authentication is done by kerberos?
+
+            // set kerberos host and realm
+            System.setProperty("java.security.krb5.realm", config.getKerberosRealm());
+            System.setProperty("java.security.krb5.kdc", config.getKerberosHost());
+
+            conf = new Configuration();
+
+            // enable kerberus
+            conf.set("hadoop.security.authentication", config.getHadoopAuthentication());
+            conf.set("hadoop.security.authorization", config.getHadoopAuthorization());
+
+            conf.set("fs.defaultFS", hdfsuri); // Set FileSystem URI
+            conf.set("fs.hdfs.impl", DistributedFileSystem.class.getName()); // Maven stuff?
+            conf.set("fs.file.impl", LocalFileSystem.class.getName()); // Maven stuff?
+
+            // hack for running locally with fake DNS records
+            // set this to true if overriding the host name in /etc/hosts
+            conf.set("dfs.client.use.datanode.hostname", config.getKerberosTestMode());
+
+            // server principal
+            // the kerberos principle that the namenode is using
+            conf.set("dfs.namenode.kerberos.principal.pattern", config.getKerberosPrincipal());
+
+            // set usergroup stuff
+            UserGroupInformation.setConfiguration(conf);
+            UserGroupInformation.loginUserFromKeytab(config.getKerberosKeytabUser(), config.getKerberosKeytabPath());
+
+            // filesystem for HDFS access is set here
+            fs = FileSystem.get(conf);
+        }
+    }
+
+    // Method for committing the AVRO-file to HDFS
+    public void commit(File syslogFile, long lastEpochMicros) {
+        // The code for writing the file to HDFS should be same for both test (non-kerberized access) and prod (kerberized access).
+        if (useMockKafkaConsumer) {
+            // CODE FOR TEST-MODE GOES HERE!
```

Comment:
Test mode as in some mock for unit tests? Questionable design if so

Reply:
Removed the mock consumer check in commit 022832c8b423efc7466b7aead56cc317a45a0002 as the method should be the same for both mock HDFS and kerberized HDFS.